pkg/controllers/edgeconnect/controller.go

```diff
@@ -132,38 +133,81 @@ func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context,
 
 	edgeConnect.ObjectMeta.Finalizers = nil
 	if err := controller.client.Update(ctx, edgeConnect); err != nil {
-		_log.Debug("updating the EdgeConnect object failed")
+		_log.Debug("reconcile deletion: updating the EdgeConnect object failed")
 
 		return errors.WithStack(err)
 	}
 
 	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
 	if err != nil {
-		_log.Debug("building EdgeConnect client failed")
+		_log.Debug("reconcile deletion: building EdgeConnect client failed")
 
 		return err
 	}
 
 	tenantEdgeConnect, err := getEdgeConnectByName(edgeConnectClient, edgeConnect.Name)
 	if err != nil {
+		_log.Debug("reconcile deletion: failed to get EdgeConnect by name")
+
 		return err
 	}
 
 	switch {
 	case tenantEdgeConnect.ID == "":
-		_log.Info("EdgeConnect not found on the tenant")
+		{
+			_log.Info("EdgeConnect not found on the tenant")
+
+			return nil
+		}
 	case !tenantEdgeConnect.ManagedByDynatraceOperator:
-		_log.Info("can't delete EdgeConnect configuration from the tenant because it has been created manually by a user")
-	case edgeConnectIdFromSecret == "":
-		_log.Info("EdgeConnect client secret is missing")
+		{
+			_log.Info("can't delete EdgeConnect configuration from the tenant because it has been created manually by a user")
 
-		return edgeConnectClient.DeleteEdgeConnect(tenantEdgeConnect.ID)
+			return nil
+		}
+	case edgeConnectIdFromSecret == "":
+		{
+			_log.Info("EdgeConnect client secret is missing")
+		}
 	default:
-		if tenantEdgeConnect.ID != edgeConnectIdFromSecret {
-			_log.Info("EdgeConnect client secret contains invalid Id")
+		{
+			if tenantEdgeConnect.ID != edgeConnectIdFromSecret {
+				_log.Info("EdgeConnect client secret contains invalid Id")
+			}
 		}
+	}
+
+	// TODO: Remove IF clause when non-provisioner mode supports Connection Settings object creation
+	if edgeConnect.IsProvisionerModeEnabled() {
+		err = controller.deleteConnectionSetting(ctx, edgeConnectClient)
+		if err != nil {
+			_log.Info("reconcile deletion: Deleting connection setting failed")
 
-		return edgeConnectClient.DeleteEdgeConnect(tenantEdgeConnect.ID)
+			return err
+		}
+	}
+
+	return edgeConnectClient.DeleteEdgeConnect(tenantEdgeConnect.ID)
+}
+
+func (controller *Controller) deleteConnectionSetting(ctx context.Context, edgeConnectClient edgeconnect.Client) error {
+	var kubeSystemNamespace corev1.Namespace
+
+	err := controller.apiReader.Get(ctx, client.ObjectKey{Name: kubeSystemNamespaceName}, &kubeSystemNamespace)
+	if err != nil {
+		return err
+	}
```

Comment:
note for future improvement:

- you could store the `kubeSystemNamespace.UID` in the `Status` of the EdgeConnect object, so you don't always do a request for it.
  - `kubeSystemNamespace.UID` does not change overtime (AFAIK), so constantly requesting it is unnecessary

Reply:
Awesome suggestion! I did it [here](https://github.com/Dynatrace/dynatrace-operator/pull/3206/commits/b16b883aa01c734fe24de6173b3c79978d4035f7)